tmt/base.py

```diff
@@ -1162,9 +1168,20 @@ def tests(self, keys=None, names=None, filters=None, conditions=None):
         conditions = (conditions or []) + list(Test._opt('conditions', []))
 
         # Build the list and convert to objects
-        return self._filters_conditions(
-            [Test(test) for test in self.tree.prune(keys=keys, names=names)],
-            filters, conditions)
+        # If duplicate test names are allowed, match test name/regexp one-by-one
+        # and preserve the order of tests within a plan
+        if unique == False and names:
```

Comment:
```suggestion
        if not unique and names:
```

Reply:
I believe all of the above is addressed in the latest PR update.